### PR TITLE
Feat: convert /admin/regions to typescript

### PR DESCRIPTION
**Pull request by pKorsholm**

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/regions/list-regions.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  @Type(() => Number)
+  offset?: number
+}
+
+export class AdminGetRegionsRes {
+  regions: Region[]
+  count: number
+  offset: number
+  limit: number
+}
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/regions/list-regions.ts:81]: Should be in `/index.ts` and be `AdminRegionsListRes`

File: packages/medusa/src/api/routes/admin/regions/remove-payment-provider.ts [hunk 2]
```diff
@@ -1,12 +1,17 @@
-import { defaultRelations, defaultFields } from "./"
+import { defaultAdminRegionRelations, defaultAdminRegionFields } from "."
+import { validator } from "../../../../utils/validator"
+import { Region } from "../../../.."
+import RegionService from "../../../../services/region"
+import { IsString } from "class-validator"
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/regions/remove-payment-provider.ts:5]: nit: move imports from external modules to the top

File: packages/medusa/src/api/routes/admin/regions/set-metadata.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { defaultAdminRegionFields, defaultAdminRegionRelations } from "."
+import { validator } from "../../../../utils/validator"
+import { Region } from "../../../.."
+import RegionService from "../../../../services/region"
+import { IsString } from "class-validator"
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/regions/set-metadata.ts:5]: nit

File: packages/medusa/src/api/routes/admin/regions/update-region.ts [hunk 4]
```diff
@@ -1,11 +1,15 @@
-import { MedusaError, Validator } from "medusa-core-utils"
-import { defaultRelations, defaultFields } from "./"
+import { defaultAdminRegionRelations, defaultAdminRegionFields } from "."
+import { validator } from "../../../../utils/validator"
+import { Region } from "../../../.."
+import RegionService from "../../../../services/region"
+import { IsArray, IsNumber, IsOptional, IsString } from "class-validator"
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/regions/update-region.ts:5]: nit

File: packages/medusa/src/api/routes/admin/regions/index.ts [hunk 5]
```diff
@@ -83,8 +84,18 @@ export const defaultFields = [
...
   "countries",
   "payment_providers",
   "fulfillment_providers",
 ]
+
+export class AdminRegionRes {
+  region: Region
+}
+
+export class DeleteResponse {
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/regions/index.ts:97]: We should either move this to common or name it `AdminRegionsDeleteRes`

File: packages/medusa/src/api/routes/admin/regions/index.ts [hunk 6]
```diff
@@ -83,8 +84,18 @@ export const defaultFields = [
...
 ]
 
-export const defaultRelations = [
+export const defaultAdminRegionRelations = [
   "countries",
   "payment_providers",
   "fulfillment_providers",
 ]
+
+export class AdminRegionRes {
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/regions/index.ts:93]:
```suggestion
export class AdminRegionsRes {
```

**Review by srindom — CHANGES_REQUESTED** [review]

File: packages/medusa/src/api/routes/admin/regions/list-regions.ts (hunk 1 above)
Comment (pKorsholm) [reply packages/medusa/src/api/routes/admin/regions/list-regions.ts:81]: Done, and made a composed type with `paginationres` from `common`

File: packages/medusa/src/api/routes/admin/regions/index.ts [hunk 7]
```diff
@@ -83,8 +85,27 @@ export const defaultFields = [
...
   "countries",
   "payment_providers",
   "fulfillment_providers",
 ]
+
+export class AdminRegionsRes {
+  region: Region
+}
+
+export type AdminGetRegionsRes = PaginatedResponse & {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/regions/index.ts:98]:
```suggestion
export type AdminRegionsListRes = PaginatedResponse & {
```

File: packages/medusa/src/api/routes/admin/regions/index.ts [hunk 8]
```diff
@@ -83,8 +85,27 @@ export const defaultFields = [
...
+  regions: Region[]
+}
+
+export type AdminDeleteRegionRes = DeleteResponse
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/regions/index.ts:102]:
```suggestion
export type AdminRegionsDeleteRes = DeleteResponse
```

**Review by olivermrbl — CHANGES_REQUESTED** [review]

**Review by olivermrbl — APPROVED** [review]
LGTM

**Pull request merged**